### PR TITLE
docs(routing): Adds RST docs on routing

### PR DESCRIPTION
**Pull request by mrclay**
Follow up of #7748
- [ ] determine if I can delete the existing Page Handler page or redirect it?
- [ ] rebase 1.x

### CODE REVIEW

**Comment by mrclay** [comment]
I missed that there's already [page handler docs](http://learn.elgg.org/en/1.10/guides/pagehandler.html).

**Comment by mrclay** [comment]
I added an "overview" section so the new Routing page has everything you need to know. Can I get the "Page handler" page to redirect to Routing? cc/ @ewinslow @Srokap

**Comment by ewinslow** [comment]
Instead of redirects, I was just leaving them out of the toc and then leaving a stub link that points to the new canonical page.

**Comment by ewinslow** [comment]
E.g. https://raw.githubusercontent.com/Elgg/Elgg/1.10/docs/about/contributing.rst

File: docs/guides/routing.rst [hunk 1]
```diff
@@ -1,4 +1,121 @@
...
+
+The ``route`` Plugin Hook
+=========================
+
+The ``route`` plugin hook is triggered earlier, before page handlers are called. The URL
+identifier is given as the type of the hook. This hook can be used to modify the identifier
+or segments, to take over page rendering completely, or just to add some logic before the
+request is handled elsewhere.
+
+Generally devs should use a page handler unless they need to affect a single page or a wider variety of URLs.
```
Comment (ewinslow) [inline docs/guides/routing.rst]: I don't think we need to make this recommendation. People can use whichever works best for them.

**Comment by ewinslow** [comment]
This is really nice. I say we merge and keep iterating. WDYT @mrclay?

**Pull request merged**